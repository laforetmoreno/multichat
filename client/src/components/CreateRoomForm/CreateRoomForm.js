import React from 'react';
import { func, string } from 'prop-types';

import Button from '../../components/Button';
import Input from '../../components/Input';
import Form from '../../components/Form';

import styles from './CreateRoomForm.module.scss';

const CreateRoomForm = ({ onSubmit, ...props }) => (
  <Form onSubmit={onSubmit} className={styles.wrapper}>
    <Input placeholder="Type the room name" {...props} />
    <Button onClick={onSubmit} className={styles.btn}>
      Add room
    </Button>
  </Form>
);

CreateRoomForm.propTypes = {
  onChange: func,
  onClick: func,
  inputValue: string,
};

export default CreateRoomForm;
