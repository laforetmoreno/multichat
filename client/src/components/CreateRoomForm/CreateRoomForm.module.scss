.wrapper {
  display: flex;
  flex-direction: column;
  margin-top: 100px;

  .btn {
    margin-top: 20px;
  }
}
