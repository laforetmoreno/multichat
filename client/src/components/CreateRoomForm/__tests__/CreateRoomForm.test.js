import React from 'react';
import CreateRoomForm from '../CreateRoomForm';
import { render, waitForElement, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';

describe('<CreateRoomForm />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<CreateRoomForm />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('testing events', async () => {
    const { getByTestId } = render(<CreateRoomForm />)
      const fieldInput = await waitForElement(() => getByTestId('input'))
      const value = 'test value'
      fireEvent.change(fieldInput, { target: { value } })

      const fieldBtn = await waitForElement(() => getByTestId('btn'))
      fireEvent.submit(fieldBtn)

      expect(fieldInput.value).toEqual(value)
  });
});