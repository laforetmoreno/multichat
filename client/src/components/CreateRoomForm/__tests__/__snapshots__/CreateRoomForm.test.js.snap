// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateRoomForm /> renders correctly 1`] = `
<form
  className="wrapper"
>
  <input
    className="wrapper "
    data-testid="input"
    placeholder="Nome da sala"
  />
  <button
    className="btn primary"
    data-testid="btn"
  >
    Criar sala
  </button>
</form>
`;
